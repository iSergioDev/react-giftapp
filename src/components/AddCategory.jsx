import React, { useState } from 'react'
import PropTypes from 'prop-types'

function AddCategory({ setCategories }) {
    const [inputValue, setInputValue] = useState('')

    const handleInputChange = (e) => {
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        
        if(inputValue.trim().length > 2){
            setCategories(categorias => [inputValue, ...categorias])
            setInputValue('')
        }
    }

  return (
    <form id='formulario' onSubmit={handleSubmit}>
        <input
            id='search'
            type="text"
            placeholder='Nueva Categoria'
            value={inputValue}
            onChange={handleInputChange}
        />
    </form>
  )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory
