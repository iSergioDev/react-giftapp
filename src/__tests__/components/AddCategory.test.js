import React from 'react'
import '@testing-library/jest-dom'
import {shallow} from 'enzyme'
import AddCategory from '../../components/AddCategory'



describe('Pruebas en <AddCategory />', () => {

    // const setCategories = () => {}
    const setCategories = jest.fn()
    let wrapper = shallow( <AddCategory setCategories={ setCategories }/>)

    beforeEach( () => {
        jest.clearAllMocks()
        wrapper = shallow( <AddCategory setCategories={ setCategories }/>) 
    })
    
    test('Debe de demostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot()
    })

    test('Debe cambiar la caja de texto del input', () => {
        const input = wrapper.find('input')
        const value = 'Hola Mundo'
        input.simulate('change', {target: {value}})
    })

    test('No debe de enviar la informaciÃ³n con submit', () => {

        wrapper.find('form').simulate('submit', { preventDefault(){}})

        expect( setCategories ).not.toHaveBeenCalled()
    })

    test('Debe de llamar el setCategories y limpiar la caja de texto', () => {
        
        const value = 'Naruto'
        // 1. simular el inputChange
        wrapper.find('input').simulate('change', {target: {value}})
        // 2. simular el submit
        wrapper.find('form').simulate('submit', { preventDefault(){}})
        // 3. setCategories se debe de haber llamado 
        expect( setCategories ).toHaveBeenCalled()
        expect( setCategories ).toHaveBeenCalledTimes(1)
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function))
        // 4. el valor de input debe de estar ''
        expect( wrapper.find('input').prop('value')).toBe('')

    })
})