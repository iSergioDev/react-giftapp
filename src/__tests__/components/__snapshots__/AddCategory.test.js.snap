// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> Debe de demostrarse correctamente 1`] = `
<form
  id="formulario"
  onSubmit={[Function]}
>
  <input
    id="search"
    onChange={[Function]}
    placeholder="Nueva Categoria"
    type="text"
    value=""
  />
</form>
`;
